src/hash/rescue/rpx/mod.rs

```diff
@@ -88,9 +91,16 @@ impl Hasher for Rpx256 {
         // every time the rate range is filled, a permutation is performed. if the final value of
         // `i` is not zero, then the chunks count wasn't enough to fill the state range, and an
         // additional permutation must be performed.
+        let mut current_element = 0_usize;
+        // handle the case of an empty `bytes`
+        let last_element = if num_field_elem == 0 {
+            current_element
+        } else {
+            num_field_elem - 1
+        };
         let i = bytes.chunks(BINARY_CHUNK_SIZE).fold(0, |i, chunk| {
             // copy the chunk into the buffer
-            if i != num_field_elem - 1 {
+            if current_element != last_element {
```

Comment:
Could we change this condition into something like:
```Rust
if chunk.len() == BINARY_CHUNK_SIZE {
```
Then, we won't need current/last element variables (and the tests seems to pass).

Reply:
I went ahead and changed but then realized that we actually want to pad with a 1 always. The case of having a final chunk which is of size 7 would not be covered by the above change. 
I added a test to illustrate when things break.

Reply:
Great catch!